### DISCUSSION TITLE
Live Update: self-hosted – is there a way to check updates?

### DISCUSSION THREAD

**Discussion by Aymkdn** [capawesome-team/capacitor-plugins#239] [Q&A]
I went through the documentation, but I cannot find how to check if there is an online update when it's self-hosted.

Should I implement my own function to verify if a newer version of the app is available?

**Comment by robingenz** [comment]
Yes, if you choose the self hosted option you need to handle that yourself.

**Reply by Aymkdn** [reply]
Thanks @robingenz for your answer.

I'm doing the below:
  1. I call `LiveUpdate.ready()`
  2. I get the online version number
  3. I use `await LiveUpdate.getBundleId()` to get the current `bundleId`
  4. If `bundleId` is null, then I use `await LiveUpdate.getVersionCode()` to get the default bundle Id
  5. I compare the online and local versions, and if there is a new one, I delete it with `await LiveUpdate.deleteBundle({ bundleId: this.newVersion })` in case it was already there for some reasons
  6. I download the bundle with `await LiveUpdate.downloadBundle({url: 'https://example.com/bundles/'+this.newVersion+'.zip', bundleId: this.newVersion })`
  7. Once downloaded, I define the new bundle Id with `await LiveUpdate.setBundle({ bundleId: this.newVersion })`
  8. Finally I reload with `await LiveUpdate.reload()`

Let me know if something is incorrect there :-) At least, it seems to work!

**Reply by robingenz** [reply]
> If bundleId is null, then I use await LiveUpdate.getVersionCode() to get the default bundle Id

`getVersionCode()` returns the native app version code. It's not related to the web bundles used by the app. The default (built-in) bundle does not has an id. 

> I compare the online and local versions

You must make sure that this also checks if the online version is compatible with the current app version. Please take a look at [What are "Binary Compatible Changes"?](https://capawesome.io/cloud/live-updates/faq/#what-are-binary-compatible-changes)
